src/server/messages.ts

```diff
@@ -29,7 +29,7 @@ export function validate(filename: string, project: string, text: string): Serve
     const encodedText = zip.toBuffer().toString();
     // TODO replace this with appropriate commands and reponse ID params
     const fullMessage = createMultipart("atf", filename, project, encodedText,
-                                        "responseID");
+                                        null);
```

Comment:
To be clear, was the problem that `"responseID"` is a fake response, and we were giving to the server something wrong?

Reply:
The communication with the server has several steps which involve similar messages. At the first stage (here), we send the ATF text (as a zip file) and get back a response ID. Later, we send a similar message with that response ID but without the text.

The `createMultipart` function is used for both due to the similar structure. It only includes the zipped ATF text if the response ID argument is `null`. Initially it only handled the first case and the argument didn't exist. When it was changed to cover both cases (with and without ID/text), the value for this argument was mistakenly (I believe!) set to a string and not null.

Reply:
I think I probably was the one setting it to a string :D